include/swift/AST/SyntaxASTMap.h

```diff
@@ -35,7 +35,7 @@ namespace syntax {
 /// a mapping from lib/AST nodes to lib/Syntax nodes while we integrate
 /// the infrastructure into the compiler.
 class SyntaxASTMap final {
-  friend class LegacyASTTransformer;
+  friend class swift::syntax::LegacyASTTransformer;
```

Comment:
The right way to fix this might be to just use `friend LegacyASTTransformer` without the `class`. Or that might not work, in which case this is okay.